lib/tvdbapiv2/rest.py

```diff
@@ -51,125 +31,50 @@
-        """
-        method = method.upper()
-        assert method in ['GET', 'HEAD', 'DELETE', 'POST', 'PUT', 'PATCH', 'OPTIONS']
-
-        if post_params and body:
-            raise ValueError(
-                "body parameter cannot be used with post_params parameter."
-            )
+    def __init__(self, session=None):
+        self.session = session
 
-        post_params = post_params or {}
-        headers = headers or {}
-
-        if 'Content-Type' not in headers:
-            headers['Content-Type'] = 'application/json'
+    def request(self, method, url, query_params=None, headers=None, body=None, post_params=None):
 
         try:
             # For `POST`, `PUT`, `PATCH`, `OPTIONS`
             if method in ['POST', 'PUT', 'PATCH', 'OPTIONS']:
-                if query_params:
-                    url += '?' + urlencode(query_params)
                 if headers['Content-Type'] == 'application/json':
-                    r = self.pool_manager.request(method, url,
-                                                  body=json.dumps(body),
-                                                  headers=headers)
+                    r = self.session.request(method, url, params=query_params, headers=headers, json=body)
                 if headers['Content-Type'] == 'application/x-www-form-urlencoded':
-                    r = self.pool_manager.request(method, url,
-                                                  fields=post_params,
-                                                  encode_multipart=False,
-                                                  headers=headers)
+                    r = self.session.request(method, url, params=query_params, headers=headers, json=post_params)
                 if headers['Content-Type'] == 'multipart/form-data':
-                    # must del headers['Content-Type'], or the correct Content-Type
-                    # which generated by urllib3 will be overwritten.
-                    del headers['Content-Type']
-                    r = self.pool_manager.request(method, url,
-                                                  fields=post_params,
-                                                  encode_multipart=True,
-                                                  headers=headers)
+                    r = self.session.request(method, url, params=query_params, headers=headers)
             # For `GET`, `HEAD`, `DELETE`
             else:
-                r = self.pool_manager.request(method, url,
-                                              fields=query_params,
-                                              headers=headers)
-        except urllib3.exceptions.SSLError as e:
+                r = self.session.request(method, url, params=query_params, headers=headers)
+
+        except RequestException as e:
             msg = "{0}\n{1}".format(type(e).__name__, str(e))
             raise ApiException(status=0, reason=msg)
 
         r = RESTResponse(r)
```

Comment:
Well since you have already changed the swagger compiled code (by adding the sessions, etc) I see no reason in retaining this.